multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,330 @@
+                    # Recompute centroids using samples from each cluster
+                    mask = (partitions[pre_view] == cl)
+                    if (np.sum(mask) == 0):
+                        new_centers.append(centroids[view][cl])
+                    else:
+                        cent = np.mean(Xs[view][mask], axis=0)
+                        new_centers.append(cent)
+                centroids[view] = np.vstack(new_centers)
+
+                # Compute expectation
+                distances = cdist(Xs[view], centroids[view])
+                partitions[view] = np.argmin(distances, axis=1).flatten()
+                # Recompute the objective function
+                o_funct = 0
+                for cl in range(self.n_clusters):
+                    # Collect data points in each cluster and compute within
+                    # cluster distances
+                    vecs = Xs[view][(partitions[view] == cl)]
+                    dist = np.linalg.norm(vecs - centroids[view][cl], axis=1)
+                    o_funct += np.sum(dist)
+
+                # Track of number of iterations without improvement
+                if(o_funct < objective[view]):
+                    objective[view] = o_funct
+                    iter_stall = 0
+                else:
+                    iter_stall += 1
+
+            # Update min_intertia and best centroids if lower intertia
+            total_inertia = np.sum(objective)
+            if(total_inertia < min_inertia or best_centroids is None):
+                min_inertia = total_inertia
+                best_centroids = centroids
+
+        # Compute final cluster centroids
+        self._final_centroids(Xs, best_centroids)
+
+        return self
+
+    def predict(self, Xs):
+
+        '''
+        Predict the cluster labels for the data.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            This list must be of size 2, corresponding to the two
+            views of the data. The two views can each have a different
+            number of features, but they must have the same number of samples.
+
+        Returns
+        -------
+        predictions : array-like, shape (n_samples,)
+            The predicted cluster labels for each sample.
+
+        '''
+
+        Xs = check_Xs(Xs, enforce_views=2)
+
+        if self.centroids_ is None:
+            msg = 'This MultiviewKMeans instance is not fitted yet.'
+            raise NotFittedError(msg)
+
+        if self.centroids_[0] is None:
+            msg = 'This MultiviewKMeans instance has no cluster centroids.'
+            raise AttributeError(msg)
+
+        dist1 = cdist(Xs[0], self.centroids_[0])
+        dist2 = cdist(Xs[1], self.centroids_[1])
+        dist_metric = dist1 + dist2
+        predictions = np.argmin(dist_metric, axis=1).flatten()
+
+        return predictions
+
+    def fit_predict(self, Xs):
```

Comment:
this should be in base I think

Reply:
Do you mean the entire implementation for the fit_predict method should be in base class?